### PR TITLE
Keyboard navigation and local rebase

### PR DESCRIPTION
**Pull request by vincentbavitz**

### CODE REVIEW

File: ts/components/session/SessionPasswordModal.tsx [hunk 1]
```diff
@@ -31,6 +31,17 @@ export class SessionPasswordModal extends React.Component<Props, State> {
...
     this.setPassword = this.setPassword.bind(this);
     this.closeDialog = this.closeDialog.bind(this);
+
+    this.onKeyUp = this.onKeyUp.bind(this);
+    window.addEventListener('keyup', this.onKeyUp);
+  }
+
+  public componentDidMount() {
+    setTimeout(() => $('#password-modal-input').focus(), 100);
+  }
```
Comment (Bilb) [inline ts/components/session/SessionPasswordModal.tsx]: Looks very hacky to me. No way to make it work with refs and without the setTimeout?  something like this: https://stackoverflow.com/a/28890330

File: ts/components/session/SessionPasswordModal.tsx [hunk 2]
```diff
@@ -31,6 +31,17 @@ export class SessionPasswordModal extends React.Component<Props, State> {
 
     this.setPassword = this.setPassword.bind(this);
     this.closeDialog = this.closeDialog.bind(this);
+
+    this.onKeyUp = this.onKeyUp.bind(this);
+    window.addEventListener('keyup', this.onKeyUp);
```
Comment (Bilb) [inline ts/components/session/SessionPasswordModal.tsx]: No need for a global window eventlistener, we are only interested in events in the password inputs

File: ts/components/session/SessionPasswordModal.tsx [hunk 3]
```diff
@@ -182,4 +193,17 @@ export class SessionPasswordModal extends React.Component<Props, State> {
       this.props.onClose();
     }
   }
+
+  private async onKeyUp(event: any) {
+    const { onOk } = this.props;
+    const setPasswordFocussed =
+      $('#password-modal-input').is(':focus') ||
+      $('#password-modal-input-confirm').is(':focus');
```
Comment (Bilb) [inline ts/components/session/SessionPasswordModal.tsx]: If you register your `this.onKeyUp()` as a callback of both of your inputs (`onKeyUp={this.onKeyUp}`) you can avoid those jquery calls

File: ts/components/session/SessionPasswordPrompt.tsx [hunk 4]
```diff
@@ -31,9 +31,17 @@ export class SessionPasswordPrompt extends React.PureComponent<{}, State> {
     window.addEventListener('keyup', this.onKeyUp);
   }
 
+  public componentDidMount() {
+    setTimeout(() => $('#password-prompt-input').focus(), 100);
```
Comment (Bilb) [inline ts/components/session/SessionPasswordPrompt.tsx]: Looks very hacky to me

File: ts/components/session/settings/SessionSettings.tsx [hunk 5]
```diff
@@ -68,9 +68,14 @@ export class SettingsView extends React.Component<SettingsViewProps, State> {
 
     this.hasPassword();
     this.refreshLinkedDevice = this.refreshLinkedDevice.bind(this);
+
+    this.onKeyUp = this.onKeyUp.bind(this);
+    window.addEventListener('keyup', this.onKeyUp);
```
Comment (Bilb) [inline ts/components/session/settings/SessionSettings.tsx]: same, should be an eventListener on the password input only

File: ts/components/session/settings/SessionSettings.tsx [hunk 6]
```diff
@@ -68,9 +68,14 @@ export class SettingsView extends React.Component<SettingsViewProps, State> {
...
   }
 
   public componentDidMount() {
+    setTimeout(() => $('#password-lock-input').focus(), 100);
```
Comment (Bilb) [inline ts/components/session/settings/SessionSettings.tsx]: Looks very hacky to me

File: ts/components/session/SessionPasswordModal.tsx [hunk 7]
```diff
@@ -31,6 +31,17 @@ export class SessionPasswordModal extends React.Component<Props, State> {
...
+    this.onKeyUp = this.onKeyUp.bind(this);
+    window.addEventListener('keyup', this.onKeyUp);
+  }
+
+  public componentDidMount() {
+    setTimeout(() => $('#password-modal-input').focus(), 100);
+  }
+
+  public componentWillUnmount() {
+    window.removeEventListener('keyup', this.onKeyUp);
```
Comment (Bilb) [inline ts/components/session/SessionPasswordModal.tsx]: to remove if handler is set in in the input

File: ts/components/session/SessionPasswordPrompt.tsx [hunk 8]
```diff
@@ -31,9 +31,17 @@ export class SessionPasswordPrompt extends React.PureComponent<{}, State> {
...
+  }
+
+  public componentWillUnmount() {
+    window.removeEventListener('keyup', this.onKeyUp);
```
Comment (Bilb) [inline ts/components/session/SessionPasswordPrompt.tsx]: to remove if handler is set in in the input

Comment (Bilb) [reply ts/components/session/SessionPasswordPrompt.tsx]: Also, the 
`     window.addEventListener('keyup', this.onKeyUp);`
should probably not be in the constructor, rather in the `componentDidMount` method instead

File: ts/components/session/settings/SessionSettings.tsx [hunk 9]
```diff
@@ -576,4 +581,14 @@ export class SettingsView extends React.Component<SettingsViewProps, State> {
         });
       });
   }
+
+  private async onKeyUp(event: any) {
+    const lockPasswordFocussed = $('#password-lock-input').is(':focus');
```
Comment (Bilb) [inline ts/components/session/settings/SessionSettings.tsx]: You can remove those if listener in set on the password field only

File: ts/components/session/SessionPasswordModal.tsx (hunk 2 above)
Comment (vincentbavitz) [reply ts/components/session/SessionPasswordModal.tsx]: All ``.tsx`` files store event listeners globally.

File: ts/components/session/SessionPasswordModal.tsx (hunk 1 above)
Comment (vincentbavitz) [reply ts/components/session/SessionPasswordModal.tsx]: Doesnt work with our current implementation.

**Pull request merged**